From bf842d0cf9ae13408e20963c977d3c85642a69b3 Mon Sep 17 00:00:00 2001
From: "Kaleb S. KEITHLEY" <kkeithle@redhat.com>
Date: Thu, 18 Jun 2020 09:38:13 -0400
Subject: [PATCH] systemd: revised drop-in conf file handling

see https://bugzilla.redhat.com/show_bug.cgi?id=1848208

systemd service file documentation still sucks rocks

This is my current best guess, and at least it doesn't log errors

Change-Id: Iec9847356090e561bfcd5f41a64b35c2e686a3d6
Signed-off-by: Kaleb S. KEITHLEY <kkeithle@redhat.com>
---
 src/nfs-ganesha.spec-in.cmake                         | 6 +++---
 src/scripts/systemd/nfs-ganesha-lock.service.debian10 | 2 +-
 src/scripts/systemd/nfs-ganesha-lock.service.el8      | 2 +-
 src/scripts/systemd/rpc-statd.conf.debian10           | 2 +-
 src/scripts/systemd/rpc-statd.conf.el8                | 2 +-
 5 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/src/nfs-ganesha.spec-in.cmake b/src/nfs-ganesha.spec-in.cmake
index 9cfda6c0e..d66106d6f 100644
--- a/src/nfs-ganesha.spec-in.cmake
+++ b/src/nfs-ganesha.spec-in.cmake
@@ -557,13 +557,13 @@ install -m 644 config_samples/vfs.conf %{buildroot}%{_sysconfdir}/ganesha
 
 mkdir -p %{buildroot}%{_unitdir}
 %if ( 0%{?fedora} ) || ( 0%{?rhel} && 0%{?rhel} >= 8 )
-mkdir -p %{buildroot}%{_sysconfdir}/systemd/system/nfs-ganesha.d
+mkdir -p %{buildroot}%{_sysconfdir}/systemd/system/nfs-ganesha-lock.service.d
 %endif
 
 install -m 644 scripts/systemd/nfs-ganesha.service.el7	%{buildroot}%{_unitdir}/nfs-ganesha.service
 %if ( 0%{?fedora} ) || ( 0%{?rhel} && 0%{?rhel} >= 8 )
 install -m 644 scripts/systemd/nfs-ganesha-lock.service.el8	%{buildroot}%{_unitdir}/nfs-ganesha-lock.service
-install -m 644 scripts/systemd/rpc-statd.conf.el8	%{buildroot}%{_sysconfdir}/systemd/system/nfs-ganesha.d/rpc-statd.conf
+install -m 644 scripts/systemd/rpc-statd.conf.el8	%{buildroot}%{_sysconfdir}/systemd/system/nfs-ganesha-lock.service.d/rpc-statd.conf
 %else
 install -m 644 scripts/systemd/nfs-ganesha-lock.service.el7	%{buildroot}%{_unitdir}/nfs-ganesha-lock.service
 %endif
@@ -677,7 +677,7 @@ exit 0
 %{_unitdir}/nfs-ganesha-lock.service
 %{_unitdir}/nfs-ganesha-config.service
 %if ( 0%{?fedora} ) || ( 0%{?rhel} && 0%{?rhel} >= 8 )
-%{_sysconfdir}/systemd/system/nfs-ganesha.d/rpc-statd.conf
+%{_sysconfdir}/systemd/system/nfs-ganesha-lock.service.d/rpc-statd.conf
 %endif
 
 %if %{with man_page}
diff --git a/src/scripts/systemd/nfs-ganesha-lock.service.debian10 b/src/scripts/systemd/nfs-ganesha-lock.service.debian10
index 67af058fe..da06ef4fe 100644
--- a/src/scripts/systemd/nfs-ganesha-lock.service.debian10
+++ b/src/scripts/systemd/nfs-ganesha-lock.service.debian10
@@ -17,7 +17,7 @@
 
 [Unit]
 Before=nfs-ganesha.service
-Conflicts=nfs-lock.service rpc-statd.service
+Conflicts=nfs-lock.service
 
 [Service]
 ExecStartPre=
diff --git a/src/scripts/systemd/nfs-ganesha-lock.service.el8 b/src/scripts/systemd/nfs-ganesha-lock.service.el8
index 67af058fe..da06ef4fe 100644
--- a/src/scripts/systemd/nfs-ganesha-lock.service.el8
+++ b/src/scripts/systemd/nfs-ganesha-lock.service.el8
@@ -17,7 +17,7 @@
 
 [Unit]
 Before=nfs-ganesha.service
-Conflicts=nfs-lock.service rpc-statd.service
+Conflicts=nfs-lock.service
 
 [Service]
 ExecStartPre=
diff --git a/src/scripts/systemd/rpc-statd.conf.debian10 b/src/scripts/systemd/rpc-statd.conf.debian10
index 2138f7467..d06fffd4c 100644
--- a/src/scripts/systemd/rpc-statd.conf.debian10
+++ b/src/scripts/systemd/rpc-statd.conf.debian10
@@ -13,7 +13,7 @@ After=nfs-config.service
 [Service]
 EnvironmentFile=-/run/sysconfig/nfs-utils
 Type=forking
-PIDFile=/var/run/rpc.statd.pid
+PIDFile=/run/rpc.statd.pid
 ExecStart=/sbin/rpc.statd --no-notify $STATDARGS
 
 [Install]
diff --git a/src/scripts/systemd/rpc-statd.conf.el8 b/src/scripts/systemd/rpc-statd.conf.el8
index 6fec09af1..b67cfdf24 100644
--- a/src/scripts/systemd/rpc-statd.conf.el8
+++ b/src/scripts/systemd/rpc-statd.conf.el8
@@ -10,5 +10,5 @@ PartOf=nfs-utils.service
 [Service]
 Environment=RPC_STATD_NO_NOTIFY=1
 Type=forking
-PIDFile=/var/run/rpc.statd.pid
+PIDFile=/run/rpc.statd.pid
 ExecStart=/usr/sbin/rpc.statd
-- 
2.26.2

